src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelManager.cs

```diff
@@ -101,9 +130,14 @@ public IEnumerable<IViewModel> ActiveViewModels
         {
             get
             {
-                lock (_managedViewModelsLock)
+                _managedViewModelsLock.EnterReadLock();
```

Comment:
Yes, that's a good solution as well. But only as extension methods, not as a helper please. We want to get rid of Helper stuff as much as possible